internal/audit/hub/hub.go

```diff
@@ -237,28 +309,40 @@ func (l *Log) streamPrefix(ctx context.Context, kind logsv1.IngestBatch_EntryKin
 			keys = make([][]byte, l.maxBatchSize)
 		} else {
 			keys = *(keysIface.(*[][]byte)) //nolint:forcetypeassert
+			if len(keys) < l.maxBatchSize {
+				keys = make([][]byte, l.maxBatchSize)
+			}
 		}
 		defer keysPool.Put(&keys)
 
 		var i int
+		var batchSizeBytes uint64
 		for it.Seek(prefix); it.ValidForPrefix(prefix); it.Next() {
 			item := it.Item()
 
+			// retrieve byte-size from key
+			size, err := strconv.ParseUint(string(item.Key()[local.KeyByteSizeStart:local.KeyByteSizeEnd]), 16, 64)
```

Comment:
Won't this fail if there are existing, unsynced keys that don't have the size encoded in the key?

Reply:
Hmm, good point. Not the easiest to switch between cases either, given the key length doesn't change 🤔

We _could_ set a new (increased?) key len, and make assertions based on the key length, but we would be putting a lot of reliance in a potentially unsafe assertion. The alternative is storing metadata in a dedicated key in Badger. 

Storing required info in the key did seem pretty neat (without the need for an additional key), though 🙂

Reply:
Hmm... I am probably missing something here. The new keys can't be the same size as the old keys, can they? The old ones were `prefix:timestamp:id` the new ones should be `prefix:timestamp:id:size` which should make them 4 bytes longer (assuming that size is a 32-bit int). 

Isn't the `GenKeyWithByteSize` function currently overwriting part of the audit ID with the message size?

Reply:
Aaah, I got horribly mixed up and treated the timestamp portion of the ULID as the whole key 🤦‍♂️. OK, I'll just add the extra bytes to the existing 20-byte key and we should be all good.